qa/rpc-tests/test_framework/mc_test/mc_test.py

```diff
@@ -27,20 +27,42 @@ def __init__(self, datadir, srcdir, ps_type = "cob_marlin"):
         assert(ps_type == "darlin" or ps_type == "cob_marlin")
         self.ps_type = ps_type
 
-    def _generate_params(self, id, circuit_type, ps_type, file_prefix, num_constraints, segment_size):
-        params_dir = self._get_params_dir(id)
+    def _get_file_prefix(self, circ_type, constant):
+        if constant != None:
+            return str(self.ps_type) + "_" + circ_type + "_"
+        return str(self.ps_type) + "_" + circ_type + "_no_const_"
 
-        if os.path.isfile(params_dir + file_prefix + "test_pk") and os.path.isfile(params_dir + file_prefix + "test_vk"):
-            return
+    def _get_args(self, op, circuit_type, constant, keyrot, ps_type, params_dir, num_constraints, segment_size, proof_path = None):
         args = []
         args.append(os.getenv("ZENDOOMC", os.path.join(self.srcdir, "zendoo/mcTest")))
-        args.append("generate")
-        args.append(str(circuit_type))
-        args.append(str(ps_type))
+        args.append(op)
+
+        args += ["-c", str(circuit_type), "-p", str(ps_type), "-s", str(segment_size), "-n", str(num_constraints)]
+
+        if constant is not None:
+            args.append("-k")
+            if constant == True:
+                args.append("CONSTANT_PLACEHOLDER")
+            else:
+                args.append(str(constant))
```

Comment:
What happens when `constant == False`?

Reply:
As you suggest, when `constant == False` you actually use a constant. That is counterintuitive, and I updated `mc_test.py` to properly handle `False` as `None` in f1aa8bb237c592621a4f635666ee5a4761bbb988. Thanks!